flink-connector-prometheus/src/main/java/org/apache/flink/connector/prometheus/sink/http/RemoteWriteResponseClassifier.java

```diff
@@ -31,11 +31,23 @@ public static boolean isRetriableErrorResponse(HttpResponse response) {
 
     public static boolean isNonRetriableErrorResponse(HttpResponse response) {
         int statusCode = response.getCode();
-        return (statusCode >= 400 && statusCode < 500) && statusCode != 429;
+        return (statusCode >= 400 && statusCode < 500)
+                && statusCode != 429
+                && !isFatalErrorResponse(response);
     }
 
     public static boolean isSuccessResponse(HttpResponse response) {
         int statusCode = response.getCode();
         return (statusCode >= 200 && statusCode < 300);
     }
+
+    public static boolean isFatalErrorResponse(HttpResponse response) {
```

Comment:
Can we merge this into "isNonRetriable"? I'm not sure what the difference is between non-retryable and fatal

Reply:
If we want, we can rename nonRetriable as Fatal

Reply:
They are very different conditions:
* non-retriable errors data errors (rejected request). The default behaviour DISCARD_AND_CONTINUE (and not configurable at the moment, as discussed)
* fatal errors are conditions that always make the operator fail. Auth failure (403) or wrong URL path (404). The behaviour is not configurable and it would not make sense to change it

Reply:
Got it. Would it be easy to make the code easier to follow? 

Maybe define an enum for the type of error rather than using a boolean. We made the mistake of fatal/notfatal in the Async sink base and it's super confusing for implementers now. Wondering if we are making the same mistake here. 

Would using an enum with FATAL, NO_RETRY, RETRY as response type help? Or is it too complicated (I'm really not sure)

Reply:
If not possible,lets at least add explanations in the javadoc for the public methods to differentiate isFatal and isNonRetriable

Reply:
Agreed. 
Refactored using an enum that covers all possible response status codes.
Refactored the callback to make it's logic clearer